source/common/upstream/upstream_impl.cc

```diff
@@ -1107,7 +1116,13 @@ ClusterInfoImpl::ClusterInfoImpl(
           config.upstream_connection_options().set_local_interface_name_on_upstream_connections()),
       added_via_api_(added_via_api), has_configured_http_filters_(false),
       per_endpoint_stats_(config.has_track_cluster_stats() &&
-                          config.track_cluster_stats().per_endpoint_stats()) {
+                          config.track_cluster_stats().per_endpoint_stats()),
+      happy_eyeballs_config_(
+          config.upstream_connection_options().has_happy_eyeballs_config()
+              ? std::make_unique<
```

Comment:
I might be wrong, but isn't `happy_eyeballs_config_` a `std::shared_ptr` and not a `unique_ptr`?

In any case, I think the only place we use `happy_eyeballs_config_`  is in `happyEyeballsConfig()`. Which is return an optional based on the pointer being null or not. How bout we make `happy_eyeballs_config_` an `optional<HappyEyeballsConfig>` (as per https://abseil.io/tips/123) and just make `happyEyeballsConfig()` do:

```
  happyEyeballsConfig() const override {
    return happy_eyeballs_config_;
  }
```

Would that work?

Reply:
done